File: Models/DSSAT_EPICST_standalone/src/java/DSSAT_EPICST_standalone/STEMP_EPIC_Exogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class STEMP_EPIC_Exogenous
{
    private Double RAIN;
    private Double DEPIR;
    private Double TMIN;
    private Double BIOMAS;
    private Double TAMP;
    private Double MULCHMASS;
    private Double TMAX;
    private Double TAV;
    private Double SNOW;
    private Double TAVG;
    
    public STEMP_EPIC_Exogenous() { }
    
    public STEMP_EPIC_Exogenous(STEMP_EPIC_Exogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.RAIN = toCopy.getRAIN();
            this.DEPIR = toCopy.getDEPIR();
            this.TMIN = toCopy.getTMIN();
            this.BIOMAS = toCopy.getBIOMAS();
            this.TAMP = toCopy.getTAMP();
            this.MULCHMASS = toCopy.getMULCHMASS();
            this.TMAX = toCopy.getTMAX();
            this.TAV = toCopy.getTAV();
            this.SNOW = toCopy.getSNOW();
            this.TAVG = toCopy.getTAVG();
        }
    }
    public Double getRAIN()
    { return RAIN; }

    public void setRAIN(Double _RAIN)
    { this.RAIN= _RAIN; } 
    
    public Double getDEPIR()
    { return DEPIR; }

    public void setDEPIR(Double _DEPIR)
    { this.DEPIR= _DEPIR; } 
    
    public Double getTMIN()
    { return TMIN; }

    public void setTMIN(Double _TMIN)
    { this.TMIN= _TMIN; } 
    
    public Double getBIOMAS()
    { return BIOMAS; }

    public void setBIOMAS(Double _BIOMAS)
    { this.BIOMAS= _BIOMAS; } 
    
    public Double getTAMP()
    { return TAMP; }

    public void setTAMP(Double _TAMP)
    { this.TAMP= _TAMP; } 
    
    public Double getMULCHMASS()
    { return MULCHMASS; }

    public void setMULCHMASS(Double _MULCHMASS)
    { this.MULCHMASS= _MULCHMASS; } 
    
    public Double getTMAX()
    { return TMAX; }

    public void setTMAX(Double _TMAX)
    { this.TMAX= _TMAX; } 
    
    public Double getTAV()
    { return TAV; }

    public void setTAV(Double _TAV)
    { this.TAV= _TAV; } 
    
    public Double getSNOW()
    { return SNOW; }

    public void setSNOW(Double _SNOW)
    { this.SNOW= _SNOW; } 
    
    public Double getTAVG()
    { return TAVG; }

    public void setTAVG(Double _TAVG)
    { this.TAVG= _TAVG; } 
    
}

File: Models/DSSAT_EPICST_standalone/src/cpp/DSSAT_EPICST_standalone/STEMP_EPIC_Exogenous.cpp
#include "STEMP_EPIC_Exogenous.h"

STEMP_EPIC_Exogenous::STEMP_EPIC_Exogenous() { }

double STEMP_EPIC_Exogenous::getTAVG() {return this-> TAVG; }
double STEMP_EPIC_Exogenous::getTAV() {return this-> TAV; }
double STEMP_EPIC_Exogenous::getTMAX() {return this-> TMAX; }
double STEMP_EPIC_Exogenous::getBIOMAS() {return this-> BIOMAS; }
double STEMP_EPIC_Exogenous::getSNOW() {return this-> SNOW; }
double STEMP_EPIC_Exogenous::getTMIN() {return this-> TMIN; }
double STEMP_EPIC_Exogenous::getDEPIR() {return this-> DEPIR; }
double STEMP_EPIC_Exogenous::getTAMP() {return this-> TAMP; }
double STEMP_EPIC_Exogenous::getMULCHMASS() {return this-> MULCHMASS; }
double STEMP_EPIC_Exogenous::getRAIN() {return this-> RAIN; }

void STEMP_EPIC_Exogenous::setTAVG(double _TAVG) { this->TAVG = _TAVG; }
void STEMP_EPIC_Exogenous::setTAV(double _TAV) { this->TAV = _TAV; }
void STEMP_EPIC_Exogenous::setTMAX(double _TMAX) { this->TMAX = _TMAX; }
void STEMP_EPIC_Exogenous::setBIOMAS(double _BIOMAS) { this->BIOMAS = _BIOMAS; }
void STEMP_EPIC_Exogenous::setSNOW(double _SNOW) { this->SNOW = _SNOW; }
void STEMP_EPIC_Exogenous::setTMIN(double _TMIN) { this->TMIN = _TMIN; }
void STEMP_EPIC_Exogenous::setDEPIR(double _DEPIR) { this->DEPIR = _DEPIR; }
void STEMP_EPIC_Exogenous::setTAMP(double _TAMP) { this->TAMP = _TAMP; }
void STEMP_EPIC_Exogenous::setMULCHMASS(double _MULCHMASS) { this->MULCHMASS = _MULCHMASS; }
void STEMP_EPIC_Exogenous::setRAIN(double _RAIN) { this->RAIN = _RAIN; }